### PR TITLE
Matthewcurtis1999/vir 1878 sample creation tests

### PR DESCRIPTION
**Pull request by mattcurts**

### CODE REVIEW

File: tests/samples/test_api.py [hunk 1]
```diff
@@ -1327,7 +1157,7 @@ async def test_download_reads(
         )
 
     if error != "404_reads":
-        async with AsyncSession(pg) as session:
+        async with AsyncSession(data_layer.samples._pg) as session:
```
Comment (igboyes) [inline tests/samples/test_api.py:1160]: Just access `pg` directly. It is much clearer than diving through the private attributes of the data layer.

File: tests/samples/test_api.py [hunk 2]
```diff
@@ -857,7 +685,9 @@ async def test_finalize(
         with pytest.raises(ResourceNotFoundError):
             await get_data_from_app(client.app).uploads.get(1)
 
-        assert not (await get_row_by_id(pg, SQLSampleReads, 1)).upload
+        assert not (
+            await get_row_by_id(data_layer.samples._pg, SQLSampleReads, 1)
+        ).upload
```
Comment (igboyes) [inline tests/samples/test_api.py:690]: Doing assertions against databases should be in data layer tests. Am I missing something here?

**Review by igboyes — CHANGES_REQUESTED** [review]
Thanks. Please also update to use your snapshot extension instead of excluding timestamps from consideration.

**Comment by mattcurts** [comment]
Needed to do the force push because I servery messed up the merge above. the last 7 commits were made on a new branch off of main to ensure there would not be merge issues.

File: tests/samples/test_data.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                "length": [3237],
+                "sequences": [7091],
+            }
+        },
+    )
+
+    with pytest.raises(ResourceNotFoundError):
+        await get_data_from_app(client.app).uploads.get(1)
+
+    assert not (await get_row_by_id(data_layer.samples._pg, SQLSampleReads, 1)).upload
```
Comment (igboyes) [inline tests/samples/test_data.py:208]: * This is a mash-up of API and DL testing. You should be directly calling the data layer, not using `client`.
* Changes to the database are not verified.

File: tests/samples/test_api.py [hunk 4]
```diff
@@ -431,21 +337,10 @@ async def test_ok(
             data["group"] = group.id
 
         resp = await client.post("/samples", data)
-        body = await resp.json()
+        await resp.json()
```
Comment (igboyes) [inline tests/samples/test_api.py:340]: ?

File: tests/samples/test_data.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+            client.mongo.samples.find_one(),
+            get_row_by_id(pg, SQLUpload, 1),
+        )
+
+        assert sample == snapshot_recent(name="mongo")
+        assert await redis.lrange("jobs_create_sample", 0, -1) == [b"bf1b993c"]
+        assert upload.reserved is True
+
+
+@pytest.mark.datatest
```
Comment (igboyes) [inline tests/samples/test_data.py:177]: I don't believe this is used anywhere else.

File: tests/samples/test_data.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+async def test_finalize(
+    data_layer: DataLayer,
+    spawn_job_client,
+    tmp_path,
+    get_sample_data,
+):
+    """
+    Test that sample can be finalized using the Jobs API.
```
Comment (igboyes) [inline tests/samples/test_data.py:185]: This is a data layer test.

**Review by igboyes — CHANGES_REQUESTED** [review]

File: tests/samples/test_data.py [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+            client.mongo.samples.find_one(),
+            get_row_by_id(pg, SQLUpload, 1),
+        )
+
+        assert sample == snapshot_recent(name="mongo")
+        assert await redis.lrange("jobs_create_sample", 0, -1) == [b"bf1b993c"]
+
+
+@pytest.mark.datatest
+async def test_finalize(
```
Comment (igboyes) [inline tests/samples/test_data.py:175]: Test needs to be improved.

**Review by igboyes — CHANGES_REQUESTED** [review]

**Review by igboyes — APPROVED** [review]

**Pull request merged**